snappy/firstboot_test.go

```diff
@@ -118,8 +112,17 @@ func (s *FirstBootTestSuite) newPartMap() (map[string]Part, error) {
 	return s.partMap, s.partMapErr
 }
 
-func (s *FirstBootTestSuite) newFakeApp() *fakePart {
-	fakeMyApp := fakePart{snapType: snap.TypeApp}
+func (s *FirstBootTestSuite) newOverlord() configurator {
+	return s.fakeOverlord
+}
+
+func (s *FirstBootTestSuite) newFakeApp() *SnapPart {
+	fakeMyApp := SnapPart{
+		m: &snapYaml{
```

Comment:
This is the sort of logic that makes it nice to have tests sitting outside of the package itself, as the test basically assumes it knows how to construct a snap, initializing private fields to values of private types.

This is not a blocker, since it was already like this before, but I hope we can gradually move away from it and into tests that know less about the internals of the implementation.